### DISCUSSION TITLE
Remove padding from DataTable?

### DISCUSSION THREAD

**Discussion by TomJGooding** [Textualize/textual#2503] [Q&A] [answered]
Is it possible to remove the padding in a DataTable? 

For example, how might you display this as a solid block rather than stripes?

```python
from textual.app import App, ComposeResult
from textual.widgets import DataTable

BLOCK = "\u2588" * 2
ROWS = [[BLOCK] * 10] * 11


class ExampleApp(App):
    CSS = """
    Screen {
        align: center middle;
    }
    DataTable {
        width: auto;
    }
    """

    def compose(self) -> ComposeResult:
        yield DataTable()

    def on_mount(self) -> None:
        table = self.query_one(DataTable)
        table.show_header = False
        table.show_cursor = False
        table.add_columns(*ROWS[0])
        table.add_rows(ROWS[1:])


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Comment by davep** [comment, marked as answer]
That padding is [baked in by the looks of things](https://github.com/Textualize/textual/blob/14850d54a3f5fed878cff1ce2f5da08503a02932/src/textual/widgets/_data_table.py#L174-L180).

As to the wider question: I probably wouldn't use a `DataTable` to start with. Perhaps there's a more appropriate method of doing what you want to do here? (outside of any other context, for example, I think I'd be more inclined to roll a widget using the [line API](https://textual.textualize.io/guide/widgets/#line-api)).

**Reply by TomJGooding** [reply]
Thanks, I thought probably not possible if baked in.

I'll have another look at the Line API, remember it looked a bit daunting when I first read through the docs!

**Reply by davep** [reply]
I can honestly say it looks more daunting than it is. I've used it for 3 or so things now (both in Textual and personal projects) and I'm always delighted at how much power it delivers vs effort.
